var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1206,3 +903,350 @@ def test(self):
+        """Run after install to tell the configuration files and Makefiles
+        to use the compilers that Spack built the package with.
+
+        If this isn't done, they'll have CC and CXX set to Spack's generic
+        cc and c++. We want them to be bound to whatever compiler
+        they were built with."""
+        if is_windows:
+            return
+        kwargs = {"ignore_absent": True, "backup": False, "string": True}
+
+        filenames = [self.get_sysconfigdata_name(), self.config_vars["makefile_filename"]]
+
+        filter_file(spack_cc, self.compiler.cc, *filenames, **kwargs)
+        if spack_cxx and self.compiler.cxx:
+            filter_file(spack_cxx, self.compiler.cxx, *filenames, **kwargs)
+
+    @run_after("install")
+    def symlink(self):
+        if is_windows:
+            return
+        spec = self.spec
+        prefix = self.prefix
+
+        # TODO:
+        # On OpenSuse 13, python uses <prefix>/lib64/python2.7/lib-dynload/*.so
+        # instead of <prefix>/lib/python2.7/lib-dynload/*.so. Oddly enough the
+        # result is that Python can not find modules like cPickle. A workaround
+        # for now is to symlink to `lib`:
+        src = os.path.join(prefix.lib64, "python{0}".format(self.version.up_to(2)), "lib-dynload")
+        dst = os.path.join(prefix.lib, "python{0}".format(self.version.up_to(2)), "lib-dynload")
+        if os.path.isdir(src) and not os.path.isdir(dst):
+            mkdirp(dst)
+            for f in os.listdir(src):
+                os.symlink(os.path.join(src, f), os.path.join(dst, f))
+
+        if spec.satisfies("@3:") and spec.satisfies("+pythoncmd"):
+            os.symlink(os.path.join(prefix.bin, "python3"), os.path.join(prefix.bin, "python"))
+            os.symlink(
+                os.path.join(prefix.bin, "python3-config"),
+                os.path.join(prefix.bin, "python-config"),
+            )
+
+    @run_after("install")
+    def install_python_gdb(self):
+        # https://devguide.python.org/gdb/
+        src = os.path.join("Tools", "gdb", "libpython.py")
+        if os.path.exists(src):
+            install(src, self.command.path + "-gdb.py")
+
+    @run_after("install")
+    @on_package_attributes(run_tests=True)
+    def import_tests(self):
+        """Test that basic Python functionality works."""
+
+        spec = self.spec
+
+        with working_dir("spack-test", create=True):
+            # Ensure that readline module works
+            if "+readline" in spec:
+                self.command("-c", "import readline")
+
+            # Ensure that ssl module works
+            if "+ssl" in spec:
+                self.command("-c", "import ssl")
+                self.command("-c", "import hashlib")
+
+            # Ensure that sqlite3 module works
+            if "+sqlite3" in spec:
+                self.command("-c", "import sqlite3")
+
+            # Ensure that dbm module works
+            if "+dbm" in spec:
+                self.command("-c", "import dbm")
```

Comment:
I don't see the Windows build handling `~dbm`. If it's not supported or we don't want to handle that for now, then I think `conflicts(~dbm, when=platform=windows)` would be appropriate.

Reply:
It never has, this PR does nothing to change that behavior. The python build script on Windows has an incredibly limited CLI so many of the variants will have no impact one way or another on Windows. I added the conflict just to represent the fact that whether or not you specify the variant, you're getting dbm on Windows.